src/widget/container/mod.rs

```diff
@@ -208,16 +177,30 @@ where
     Display: DrawTarget,
 {
     fn intrinsic_size(&self) -> IntrinsicSize {
-        let content_size = self.content_size();
+        let total_size = self
+            .content_size()
+            .outset(self.options.padding)
+            .outset(self.options.margin);
 
         IntrinsicSize::new(
-            self.options.width.or(content_size.width),
-            self.options.height.or(content_size.height),
+            self.options.width.or(total_size.width),
+            self.options.height.or(total_size.height),
         )
     }
 
     fn draw(&self, display: &mut Display, origin: Point, size: Size) -> Result<(), Display::Error> {
-        self.draw_self(display, origin, size)?;
-        self.draw_children(display, origin, size)
+        let box_origin = Point::new(
+            origin.x + self.options.margin.left as i32,
+            origin.y + self.options.margin.top as i32,
+        );
+        let box_size = size.inset(self.options.margin);
+        self.draw_self(display, box_origin, box_size)?;
+
+        let content_origin = Point::new(
+            box_origin.x + self.options.padding.left as i32,
```

Comment:
In the CSS model border is between margin + padding, would that make sense here too (otherwise seems like you have to manually compensate for border in your padding values)?

Reply:
Ah yeah, good catch. Fixed (I think)